### PR TITLE
4.7 RN: Add known Kuryr issue with UPI

### PR DESCRIPTION
**Pull request by dulek**
We've discovered the problem with UPI installations with Kuryr [1] and
realized there's a simple workaround to the issue. This commit adds
information about it to the 4.7 relase notes.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1927244

### CODE REVIEW

File: release_notes/ocp-4-7-release-notes.adoc [hunk 1]
```diff
@@ -1014,6 +1014,29 @@ The root cause of this issue is unknown and no workaround currently exists.
...
+This is caused by changes in how Kuryr detects the Neutron subnet of the {product-title} nodes. The workaround is to not remove the master Machine manifests, so in step 2 of "Creating the Kubernetes manifest and Ignition config files" in installation instructions the command to use becomes:
++
+[source,terminal]
+----
+$ rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+----
++
+The issue is scheduled to be fixed in one of the z-stream {product-title} 4.7 updates.
++
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1927244[*BZ#1927244*])
```
Comment (maxwelldb) [inline release_notes/ocp-4-7-release-notes.adoc:1039]: @dulek So, I'm basically thinking this:

```suggestion
* Installing a cluster on {rh-openstack} that uses Kuryr times out with the following messages during bootstrapping: 
+
[source,terminal]
----
INFO Waiting up to 20m0s for the Kubernetes API at https://api.ostest.shiftstack.com:6443...
INFO API v1.20.0+ba45583 up
INFO Waiting up to 30m0s for bootstrapping to complete...
ERROR Attempted to gather ClusterOperator status after wait failure: listing ClusterOperator objects: Get "https://api.ostest.shiftstack.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 10.46.44.166:6443: connect: connection refused
INFO Use the following commands to gather logs from the cluster
INFO openshift-install gather bootstrap --help
FATAL failed to wait for bootstrapping to complete: timed out waiting for the condition
----

The time out is caused by changes in how Kuryr detects the {rh-openstack} Networking service (neutron) subnet of the cluster's nodes. 

As a workaround, do not remove the control plane machine manifests as described by "Creating the Kubernetes manifest and Ignition config files" in the installation documentation.When you are instructed to run the following command:
+
[source,terminal]
----
$ rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml openshift/99_openshift-cluster-api_worker-machineset-*.yaml
----

Run this command instead:
+
[source,terminal]
----
$ rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
----
```
I could see about adjusting the prod Kuryr UPI docs to use the latter command, instead. I could then revert the change after the bug is fixed. This would trigger a QE review, though, and it's not as easy to get approved after freeze as a release note is. 

Thoughts?

**Review by maxwelldb — CHANGES_REQUESTED** [review]

File: release_notes/ocp-4-7-release-notes.adoc (hunk 1 above)
Comment (dulek) [reply release_notes/ocp-4-7-release-notes.adoc:1039]: I'd rather not touch the documentation itself. The step with removing master machines manifests is correct and in the future not removing them may have adverse side effects. I don't want anyone to write this workaround into their scripts without intent.

**Comment by dulek** [comment]
Aw, I guess I need to squash this.

**Comment by dulek** [comment]
Done, wonderful suggestions BTW.

File: release_notes/ocp-4-7-release-notes.adoc (hunk 1 above)
Comment (maxwelldb) [reply release_notes/ocp-4-7-release-notes.adoc]: Works for me. Just wanted to present options. :)

**Comment by maxwelldb** [comment]
@dulek Thanks!

**Comment by maxwelldb** [comment]
Direct preview link: https://deploy-preview-29462--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-7-release-notes.html#ocp-4-7-known-issues

**Comment by MaysaMacedo** [comment]
there are a few "+" in between removal commands.

File: release_notes/ocp-4-7-release-notes.adoc [hunk 2]
```diff
@@ -1014,6 +1014,34 @@ The root cause of this issue is unknown and no workaround currently exists.
...
+----
+$ rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+----
+
+Run this command instead:
++
+[source,terminal]
+----
+$ rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+----
```
Comment (maxwelldb) [inline release_notes/ocp-4-7-release-notes.adoc:1045]:
```suggestion

(link:https://bugzilla.redhat.com/show_bug.cgi?id=1927244[*BZ#1927244*])

```

Realized I obliterated the BZ link.

File: release_notes/ocp-4-7-release-notes.adoc [hunk 3]
```diff
@@ -1014,6 +1014,36 @@ The root cause of this issue is unknown and no workaround currently exists.
...
+INFO Waiting up to 20m0s for the Kubernetes API at https://api.ostest.shiftstack.com:6443...
+INFO API v1.20.0+ba45583 up
+INFO Waiting up to 30m0s for bootstrapping to complete...
+ERROR Attempted to gather ClusterOperator status after wait failure: listing ClusterOperator objects: Get "https://api.ostest.shiftstack.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 10.46.44.166:6443: connect: connection refused
+INFO Use the following commands to gather logs from the cluster
+INFO openshift-install gather bootstrap --help
+FATAL failed to wait for bootstrapping to complete: timed out waiting for the condition
+----
+
+The time out is caused by changes in how Kuryr detects the {rh-openstack} Networking service (neutron) subnet of the cluster's nodes.
```
Comment (jeana-redhat) [inline release_notes/ocp-4-7-release-notes.adoc:1030]: s/time out/timeout

File: release_notes/ocp-4-7-release-notes.adoc [hunk 4]
```diff
@@ -1014,6 +1014,36 @@ The root cause of this issue is unknown and no workaround currently exists.
...
+INFO Waiting up to 30m0s for bootstrapping to complete...
+ERROR Attempted to gather ClusterOperator status after wait failure: listing ClusterOperator objects: Get "https://api.ostest.shiftstack.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 10.46.44.166:6443: connect: connection refused
+INFO Use the following commands to gather logs from the cluster
+INFO openshift-install gather bootstrap --help
+FATAL failed to wait for bootstrapping to complete: timed out waiting for the condition
+----
+
+The time out is caused by changes in how Kuryr detects the {rh-openstack} Networking service (neutron) subnet of the cluster's nodes. 
+
+As a workaround, do not remove the control plane machine manifests as described by the "Creating the Kubernetes manifest and Ignition config files" section in the installation documentation. When you are instructed to run the following command:
```
Comment (jeana-redhat) [inline release_notes/ocp-4-7-release-notes.adoc:1032]: I don't know if you want to, but since the RN is an assembly you can direct link to that reference if you like

**Review by jeana-redhat — COMMENTED** [review]
Small suggestions, mostly just missing some `+`s to keep everything in the bullet as mentioned in chat :)

File: release_notes/ocp-4-7-release-notes.adoc (hunk 3 above)
Comment (maxwelldb) [inline release_notes/ocp-4-7-release-notes.adoc:1030]:
```suggestion
The timeout is caused by changes in how Kuryr detects the {rh-openstack} Networking service (neutron) subnet of the cluster's nodes. 
```

File: release_notes/ocp-4-7-release-notes.adoc [hunk 5]
```diff
@@ -1014,6 +1014,36 @@ The root cause of this issue is unknown and no workaround currently exists.
...
+----
+
+Run this command instead:
++
+[source,terminal]
+----
+$ rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+----
+
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1927244[*BZ#1927244*])
```
Comment (jeana-redhat) [inline release_notes/ocp-4-7-release-notes.adoc:1046]: Also this isn't in the preview, so now I don't trust the preview :thinking:

Comment (maxwelldb) [reply release_notes/ocp-4-7-release-notes.adoc:1046]: https://media.giphy.com/media/Z8lLZNsgELPzO/giphy.gif

I'm seeing it now.

**Comment by maxwelldb** [comment]
Waiting on the new build...

**Review by jeana-redhat — APPROVED** [review]

**Review by maxwelldb — APPROVED** [review]
